docs/source/55_api.md

```diff
@@ -339,6 +342,134 @@ val bcastFeatures = sc.loadFeatures("my/features.adam").broadcast()
+left side of the join.
+
+To perform a `groupBy` after the join, BroadcastRegionJoin only supports
+grouping by the right dataset, and ShuffleRegionJoin supports only grouping by
+the left dataset. Thus, depending on the type of join, the left and right
+dataset may change between BroadcastRegionJoin and ShuffleRegionJoin.
+
+The reason BroadcastRegionJoin does not have a `joinAndGroupByLeft`
+implementation is due to the fact that the left dataset is broadcasted to all
+nodes. It would be impossible to perform a group by function on the resulting
+join with predictable performance because, unlike the ShuffleRegionJoin, we do
+not guarantee any sort invariants. Because the ShuffleRegionJoin performs a
+sort-merge join, grouping by the left data does not require a shuffle.
+
+###### Separate reads into overlapping and non-overlapping features
+
+This query joins an RDD of reads with an RDD of features using an outer join.
+The outer join will produce an RDD where each read is optionally mapped to a
+feature. If a given read does not overlap with any features provided, it is
+paired with a `None`. After we perform the join, we use a predicate to separate
+the reads into two RDDs. This query is useful for filtering out reads based on
+feature data. For example, identifying reads that overlap with ATAC-seq data to
+perform chromatin accessibility studies. It may be useful to separate the reads
+to perform distinct analyses on each resulting dataset.
+
+```scala
+// An outer join provides us with both overlapping and non-overlapping data
+val reads = sc.loadAlignments(“my/reads.adam”)
+val features = sc.loadFeatures(“my/features.adam”)
+
+// As a ShuffleRegionJoin, we can use a LeftOuterShuffleRegionJoin:
+val readsToFeatures = reads.leftOuterShuffleRegionJoin(features)
+
+// As a BroadcastRegionJoin, we can use a RightOuterBroadcastRegionJoin:
+val featuresToReads = features.rightOuterShuffleRegionJoin(reads)
+
+// After we have our join, we need to separate the RDD
+// If we used the ShuffleRegionJoin, we filter by None in the values
+val overlapsFeatures = readsToFeatures.rdd.filter(_._2.isDefined)
+val notOverlapsFeatures = readsToFeatures.rdd.filter(_._2.isEmpty)
+
+// If we used BroadcastRegionJoin, we filter by None in the keys
+val overlapsFeatures = featuresToReads.rdd.filter(_._1.isDefined)
+val notOverlapsFeatures = featuresToReads.rdd.filter(_._1.isEmpty)
+```
+
+Because of the difference in how ShuffleRegionJoin and BroadcastRegionJoin are
+called, the predicate changes between them. It is not possible to call a
+`leftOuterJoin` using the BroadcastRegionJoin. As previously mentioned, the
+BroadcastRegionJoin broadcasts the left dataset, so a left outer join would
+require an additional shuffle phase. For an outer join, using a
+ShuffleRegionJoin will be cheaper if your reads are already sorted, however if
+the feature dataset is small, the BroadcastRegionJoin call would likely be more
```

Comment:
`the feature dataset is small` -> `the feature dataset is small and the reads are not sorted`